datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -336,6 +336,19 @@ public SdxCluster getByNameInAccount(String userCrn, String name) {
         }
     }
 
+    public SdxCluster getByNameInAccountAllowDetached(String userCrn, String name) {
+        LOGGER.info("Searching for SDX cluster by name {} allowing detached", name);
+        String accountIdFromCrn = getAccountIdFromCrn(userCrn);
+        Optional<SdxCluster> sdxCluster = measure(() ->
+                        sdxClusterRepository.findByAccountIdAndClusterNameAndDeletedIsNull(accountIdFromCrn, name), LOGGER,
```

Comment:
It looks good for me, but can it happen this repo call can return with more than 1 entry? (it would cause an exception)

Reply:
Good question. I don't believe this is possible. While the repository is not constrained on name uniqueness, there are various checks that ensure the cluster does not have a name in the repo already.

Even in detach the steps are very concise and make it so the two datalakes never share a name. So this should not happen no, without manually messing with the DB.